dev/build/plugin-versions/bignums.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+case "$1" in
+    git-master|/cygdrive*)
+        # First get the SHA of master/HEAD and then download a zip of exactly this state
+        # Note: the access token is a public access token of MSoegtropIMC
+        BIGNUMS_SHA=$(curl \
+            -H 'Authorization: token 79b1bf691f933bb0185f0fc72816e5ff387bc6ea' \
```

Comment:
You shouldn't have pushed this to GitHub, it looks like a personal access token. You should revoke it ASAP. Furthermore, you don't need this to get what you are looking for. You should use https://github.com/coq/bignums/archive/master.zip instead.

Reply:
@Zimmi48 : yes it is a personal access token, but it has exactly the same access rights as public users have anyway (that means it has zero extra rights). As far as I can tell you need an access token to access the API, otherwise the number of accesses you can make from one machine is very restricted (to 50 per hour or so). With an access token you can do 5000 per hour. The worst thing which can happen is that someone uses it for something else and steals from the 5000 requests per hour, but there is no good reason for doing this.

The reason I am not using master.zip is that the downloaded files are cached. This mechanism includes the SHA of the version in the cached archive name. If a new version is available, it will be downloaded and compiled automatically. Without such a mechanism one cannot reasonably cache downloads of plugins.

Reply:
Thanks for the explanations. It looks reasonable then. I'll have a more thorough look at the PR later.

Reply:
Just a note: Since I am also not sure what is the right procedure, I tried it without a token like this:

for i in {1..100}; do curl -H 'Accept: application/vnd.github.VERSION.sha'   'https://api.github.com/repos/MSoegtropIMC/coq/commits/HEAD?sha=master'; done

The first 50 or so go fine, after this I get this error:

{
  "message": "API rate limit exceeded for <myip>. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)",
  "documentation_url": "https://developer.github.com/v3/#rate-limiting"
}

Maybe someone can review the link given above and see if there is a better way.
According to the github doc, this restriction is per hour.

Of cause I am open to better suggestions.

Reply:
If you have git installed, `git ls-remote https://github.com/coq/bignums refs/heads/master` will print (right now) `1cafe43151450d7a3f7a0b2c8036851900e3eab9        refs/heads/master`, and `git ls-remote https://github.com/coq/bignums refs/heads/master | cut -f 1` will give just the sha.

Reply:
(I ran 100 in a row without issue)

Reply:
@SkySkimmer : thanks, I will switch to the method you suggest. It also has the advantage that it is not limited to repositories in github. I didn't know that git can get such information without cloning the repo - interesting.